### DISCUSSION TITLE
怎么部署啊，搞半天没搞明白

### DISCUSSION THREAD

**Discussion by JasonNing96** [KenyonY/openai-forward#118] [Q&A]
搞了3天了，头皮发麻，感觉部署了，又感觉没部署

配置：
1. wo我在openai_key 这里放了一个 标准的key，在forward_key 这里自己写了一个
![image](https://github.com/KenyonY/openai-forward/assets/54830624/3074a02a-4f4f-4b39-8804-a1daf014eada)

2. Forward host 直接用的是这个
![image](https://github.com/KenyonY/openai-forward/assets/54830624/9d6a5f18-96a7-4869-984b-155df3ff135a)

调用：
1. 我在 chatbox的API HOST 配置这里填入了我的Public_ip 和端口号
2.
![image](https://github.com/KenyonY/openai-forward/assets/54830624/fd4a92f6-cd1c-4795-825e-d8aff1924d37)

we问题：
一直都是error，怎么填都不行。
![image](https://github.com/KenyonY/openai-forward/assets/54830624/2162d2d3-a4ae-401b-956e-ee0627055595)

求教这玩意到底怎么配置啊 啊啊啊啊啊啊

**Comment by KenyonY** [comment]
这个转发服务是需要部署在可以访问openai api的服务器上，你可以直接运行`aifd run`进行测试。

**Comment by JasonNing96** [comment]
![image](https://github.com/KenyonY/openai-forward/assets/54830624/9cd6420b-c05d-49e2-816d-891f55495e5c)
![image](https://github.com/KenyonY/openai-forward/assets/54830624/9ef1be8f-6d2f-43c0-988f-2e08002f0fc6)
啥意思？我这本地启动没问题

**Comment by JasonNing96** [comment]
但是报错 "421 Misdirected Reques“

**Comment by JasonNing96** [comment]
也能访问到，8000端口
![image](https://github.com/KenyonY/openai-forward/assets/54830624/bf9a17de-9db4-4267-afa4-544e62c51cd6)

**Comment by JasonNing96** [comment]
解决问题了，配置方法的问题，主要是 寻找的端点 是 http:IP:Port/V1，功能非常强大，就是说明不太清楚，要是大佬能补充一些文档就完美了~

**Comment by KenyonY** [comment]
确实文档写得不够详细，如果你有兴趣，欢迎贡献

**Comment by JasonNing96** [comment]
好的，非常感谢这个项目，功能非常强大，最近也写了一些配置方面的研究。

**Reply by langlibaitiaoji** [reply]
兄弟，求助，我也整了半天没整明白，想请教一下该如何配置呢。
是这样的，我开了百度千帆的coding plan，但是AI IDE里不让添加第三方模型（没办法填写API Base URL），只能选定模型厂商为openai，所以我想添加openai的自定义模型，但是实际访问时，再将请求转发至百度的接口'https://qianfan.baidubce.com/v2/coding/chat/completions'。
我的配置文件如下：
CHAT_COMPLETION_ROUTE=/v1/chat/completions
FORWARD_CONFIG=[{"base_url":"https://qianfan.baidubce.com/v2/coding/chat/completions","route":"/","type":"general"}]

我理解，是不是得先改hosts文件，加一行：127.0.0.1 api.openai.com
然后我的AI IDE访问api.openai.com时，就会重定向到本地部署的转发服务。

但是我启动项目后，自己使用curl命令测试，总是报错。
curl http://localhost:8001/v1 -H "Content-Type: application/json" -H "Authorization: Bearer xxx" -d "{\"model\": \"qianfan-code-latest\",\"messages\": [{\"role\": \"user\", \"content\": \"你好请问你是谁\"}]}"

<html><title>403: Forbidden</title><body>403: Forbidden</body></html>
